pkg/backend/identityapi/register_client.go

```diff
@@ -13,16 +14,25 @@ import (
 	"strings"
 	"time"
 
+	"github.com/antihax/optional"
 	backendhttp "github.com/newrelic/infrastructure-agent/pkg/backend/http"
 	"github.com/newrelic/infrastructure-agent/pkg/backend/inventoryapi"
 	"github.com/newrelic/infrastructure-agent/pkg/entity"
+	"github.com/newrelic/infrastructure-agent/pkg/identity-client"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/protocol"
 	"github.com/newrelic/infrastructure-agent/pkg/log"
 )
 
-var rlog = log.WithComponent("IdentityRegisterClient")
+var rlog = log.WithComponent("identityapi.RegisterClient")
 
-type IdentityRegisterClient interface {
-	Register(agentEntityID entity.ID, entities []RegisterEntity) ([]RegisterEntityResponse, time.Duration, error)
+const (
+	identityPath = "/identity/v1"
+)
+
+type RegisterClient interface {
+	RegisterEntitiesRemoveMe(agentEntityID entity.ID, entities []RegisterEntity) ([]RegisterEntityResponse, time.Duration, error)
+	RegisterProtocolEntities(agentEntityID entity.ID, entities []protocol.Entity) (RegisterBatchEntityResponse, time.Duration, error)
+	RegisterEntity(agentEntityID entity.ID, entity protocol.Entity) (RegisterEntityResponse, error)
```

Comment:
docblocks pls
btw, as 1st one is the batched flavour, could it be renamed in a way that's not misleading? ie: `RegisterEntities`

Reply:
Dam it, that was the plan when we renamed the old one 🤦